hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -352,20 +353,43 @@ object LoweredTableReader {
 
           val partitioner = new RVDPartitioner(pkType,
             sortedPartData.map { partData =>
-              Interval(selectPK(partData.getAs[Row](0)), selectPK(partData.getAs[Row](1)), includesStart = true, includesEnd = true)
-            }, key.length)
+              Interval(selectPK(partData.getAs[Row](1)), selectPK(partData.getAs[Row](2)), includesStart = true, includesEnd = true)
+            }, pkType.size)
```

Comment:
Doesn't really matter, but this is the default value of `allowedOverlap`; no need to write it explicitly, since it really means "do no extra checks".